one-off/nuke-workitem-embedded-images.sql

```diff
@@ -0,0 +1,174 @@
+            charindex('>', @richtextcol, charindex('<img', @richtextcol)) as endpos,
+            1 as rownumber
+        where charindex('<img', @richtextcol) > 0
+
+        union all
+
+        select
+            charindex('<img', @richtextcol, endpos) as startpos,
+            charindex('>', @richtextcol, charindex('<img', @richtextcol, endpos)) as endpos,
+            rownumber + 1
+        from
+            recursivecte
+        where
+            charindex('<img', @richtextcol, endpos) > 0
+    )
+    select
+        substring(
+            @richtextcol,
+            startpos,
+            endpos - startpos + 1
+        ) as imgelem,
+        convert(binary(20), '0x'+ substring(
+            @richtextcol,
+            charindex('downloadblob.img/', @richtextcol, startpos) + len('downloadblob.img/'),
+            charindex('"', @richtextcol, charindex('downloadblob.img/', @richtextcol, startpos) + len('downloadblob.img/')) - charindex('downloadblob.img/', @richtextcol, startpos) - len('downloadblob.img/')
+        ),1) as hash
+    from
+        recursivecte
+    where
+        charindex('downloadblob.img/', @richtextcol, startpos) > 0
+)
+
+GO
+
+
+create or alter function [dbo].[ReplaceBase64Images] (@RichTextCol varchar(max), @BlankBase64Image varchar(max))
+returns varchar(max)
+as
+begin
+    declare @StartIndex int = charindex('<img src="data:image/', @RichTextCol);
+
+    while @StartIndex > 0
+    begin
+        set @RichTextCol = stuff(
+            @RichTextCol,
+            @StartIndex,
+            charindex('">', substring(@RichTextCol, @StartIndex, len(@RichTextCol))) + 2,
+            @BlankBase64Image
+        );
+
+        set @StartIndex = charindex('<img src="data:image/', @RichTextCol, @StartIndex + len(@BlankBase64Image));
+    end;
+
+    return @RichTextCol;
+end;
+GO
+
+
+create or alter procedure dbo.NukeWorkitemEmbeddedImages
+    @oid varchar(20),
+	@saveChanges bit
+as
+begin
+    declare @Id int, @assetType varchar(100)
+
+	declare @error int, @rowcount int
+
+    select @Id = ba.ID, @assetType = ba.AssetType
+    from BaseAsset_Now ba
+	join dbo.OidSplit(@oid) oid on ba.ID = oid.ID and ba.AssetType = oid.AssetType
+
+    if (@@rowcount <> 1)
+    begin
+        raiserror('Invalid asset oid', 0, 1)
+        return
+    end
+
+    declare @blankimage varbinary(max) = 0x89504e470d0a1a0a0000000d4948445200000001000000010100000000376ef9240000001049444154789c626001000000ffff03000006000557bfabd40000000049454e44ae426082
+	declare @blankImageBase64 varchar(max) = CAST(N'' AS XML).value('xs:base64Binary(xs:hexBinary(sql:variable("@blankimage")))', 'VARCHAR(MAX)')
+    declare @imgpng int
+    
+    begin tran; save tran TX
+
+    exec dbo._SaveString N'img/png', @imgpng output
+
+    update Blob 
+    set Value = @blankimage, ContentType=@imgpng
+    where ID in (
+        select Content
+        from dbo.EmbeddedImage
+        where AssetID = @Id
+    )
+    and cast(Value as varbinary(max)) <> @blankimage
```

Comment:
so also update affected `EmbeddedImage.ContentType` and `EmbeddedImage_Now.ContentType` to `@imgpng`

Reply:
on second thought, we probably ought to whack these blobs as well, and update their referencing `EmbeddedImage.Content` and `EmbeddedImage_Now.Content` with `ID` of the "blank image" blob